lib/wallet/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -1895,26 +1895,22 @@ unsafeValueToLovelace v =
 -- See 'W.TokenBundleSizeAssessor' for the expected properties of this function.
 --
 tokenBundleSizeAssessor :: TokenBundleMaxSize -> TokenBundleSizeAssessor
-tokenBundleSizeAssessor maxSize = TokenBundleSizeAssessor {..}
-  where
-    assessTokenBundleSize tb
-        | serializedLengthBytes <= maxSize' =
-            TokenBundleSizeWithinLimit
-        | otherwise =
-            TokenBundleSizeExceedsLimit
-      where
-        serializedLengthBytes :: W.TxSize
-        serializedLengthBytes = computeTokenBundleSerializedLengthBytes tb
-
-        maxSize' :: W.TxSize
-        maxSize' = W.unTokenBundleMaxSize maxSize
+tokenBundleSizeAssessor maxSize =
+    TokenBundleSizeAssessor {
+        assessTokenBundleSize = \bundle ->
+            if tokenBundleSerializedLengthBytes bundle
+               <= naturalTokenBundleMaxSize maxSize
+                then TokenBundleSizeWithinLimit
+                else TokenBundleSizeExceedsLimit
+        }
 
-computeTokenBundleSerializedLengthBytes :: TokenBundle.TokenBundle -> W.TxSize
-computeTokenBundleSerializedLengthBytes = W.TxSize . safeCast
-    . BS.length . serialize' . Cardano.toMaryValue . toCardanoValue
-  where
-    safeCast :: Int -> Natural
-    safeCast = fromIntegral
+tokenBundleSerializedLengthBytes :: TokenBundle.TokenBundle -> Natural
+tokenBundleSerializedLengthBytes =
```

Comment:
My motivation to remove `TxSize` was that this type name misleads me into thinking that its a "transaction size", while this particular expression measures size of a token bundle. 

You bring a concern of expressing a unit of measurement (bytes) and I like the idea of capturing it a type level in order not to have to rely on naming convention to convey this meaning. This aspect, however, wasn't captured by the `TxSize` type name so I don't think that change to `Natural` makes it worth.

Suggestion: if we want to distinguish "bytes" from, say, "bits" on the type level then I suggest us to use existing primitive type `Quantity "byte" Natural`.

Reply:
> You bring a concern of expressing a unit of measurement (bytes) and I like the idea of capturing it a type level in order not to have to rely on naming convention to convey this meaning. This aspect, however, wasn't captured by the `TxSize` type name

I agree that the naming of `TxSize` could possibly be improved.

However since that type is already in use for the sizes of other parts of transactions, I would propose that:
- in **this** PR, we retain the use of `TxSize` for (for consistency with other measurements of sizes of transaction parts).
- in a **subsequent** PR, we tackle the issue of what we want to do with `TxSize`. Perhaps we could simply rename it to `TxSizeBytes`.

In particular, I'd suggest that these be separate PRs.

How does that sound?

(I'm slightly nervous about using `Quantity` to replace `TxSize`, as I think this will also require us to change the definitions of generators and `Arbitrary` instances, as the `TxSize` type is used in quite a few places. I'd want to make sure that we didn't inadvertently cause a drop in test coverage.)